src/Symfony/Component/Security/Http/Tests/Authentication/AuthenticationUtilsTest.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Tests\Authentication;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Security\Http\Authentication\AuthenticationUtils;
+
+class AuthenticationUtilsTest extends TestCase
+{
+    public function testLastAuthenticationErrorWhenRequestHasAttribute()
+    {
+        $request = Request::create('/');
+        $request->attributes->set(Security::AUTHENTICATION_ERROR, 'my error');
+
+        $requestStack = new RequestStack();
+        $requestStack->push($request);
+
+        $utils = new AuthenticationUtils($requestStack);
+        $this->assertSame('my error', $utils->getLastAuthenticationError());
+    }
+
+    /**
+     * @runInSeparateProcess
```

Comment:
Is this here (and in the other tests) really needed?

Reply:
Yes, because we're dealing with sessions. If we don't add these, sessions overlap and it's a bit of a mess 😄